### DISCUSSION TITLE
Should CRSs be stored as URIs or URNs?

### DISCUSSION THREAD

**Discussion by AndrewAnnex** [developmentseed/morecantile#96] [Q&A]
according to the v1 tms spec, it looks like CRSs should be stored as URIs rather than as URNs, but all of the default CRSs included in morecantile use URNs for the crs/supported crs fields, except for the wellknownscalesets

When saving a TMS to json only URIs are used.

**Comment by vincentsarago** [comment]
some background: https://github.com/developmentseed/morecantile/issues/60

we used urn because of weird behaviour when using URI. 

That's true that the `TileMatrixSet().json` will return URI (https://github.com/developmentseed/morecantile/blob/main/morecantile/models.py#L72-L84 🙁

**Reply by AndrewAnnex** [reply]
that's interesting that gdal vs proj had different support for https vs http in the URI, looking through the PROJ src now 2 years later it looks like https prefixes are supported (https://github.com/OSGeo/PROJ/blob/c41e7a6452a9d2318967c50a644ba8190021d14c/src/iso19111/io.cpp#L6703-L6707).

I added the following lines to https://github.com/AndrewAnnex/planetcantile/pull/5  to update the json strings to use the URNs. Now my unit test is passing that creates the TMS objects from the json files which is good.

For that issue I've also opened https://github.com/OSGeo/PROJ/issues/3568 to see if support for the IAU URIs can be addressed.


```python
def CRS_to_info(crs: CRS)-> tuple[str]:
    """Get CRS Info"""
    authority = "EPSG"
    code = "4326"
    version = "0"
    # attempt to grab the authority, version, and code from the CRS
    authority_code = crs.to_authority(min_confidence=20)
    if authority_code is not None:
        authority, code = authority_code
        # if we have a version number in the authority, split it out
        if "_" in authority:
            authority, version = authority.split("_")
    return authority, version, code


def CRS_to_uri(crs: CRS) -> str:
    """Convert CRS to URI."""
    authority, version, code = CRS_to_info(crs)
    return f"http://www.opengis.net/def/crs/{authority}/{version}/{code}"


def CRS_to_urn(crs: CRS)-> str:
    """Convert CRS to URN."""
    authority, version, code = CRS_to_info(crs)
    if version == '0':
        version = ''
    return f"urn:ogc:def:crs:{authority}:{version}:{code}"
```
